daemon/build.go

```diff
@@ -19,25 +20,47 @@ type releaseableLayer struct {
 	layerStore layer.Store
 	roLayer    layer.Layer
 	rwLayer    layer.RWLayer
+	newLayer   layer.Layer
 }
 
 func (rl *releaseableLayer) Mount() (string, error) {
-	if rl.roLayer == nil {
-		return "", errors.New("can not mount an image with no root FS")
-	}
 	var err error
+	var chainID layer.ChainID
+	if rl.roLayer != nil {
+		chainID = rl.roLayer.ChainID()
+	}
+
 	mountID := stringid.GenerateRandomID()
-	rl.rwLayer, err = rl.layerStore.CreateRWLayer(mountID, rl.roLayer.ChainID(), nil)
+	rl.rwLayer, err = rl.layerStore.CreateRWLayer(mountID, chainID, nil)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to create rwlayer")
 	}
 
 	return rl.rwLayer.Mount("")
 }
 
+func (rl *releaseableLayer) Commit() (layer.DiffID, error) {
```

Comment:
Ok, I thought `layer.DiffID` was probably a bad return value, but wanted to keep it as simple as possible for now.

How should the `DiffID` of the new layer be exposed? Should `releaseableLayer` has a method `DiffID()` ?